### ISSUE TITLE
[ML/SXR] Extend rotational tomography to multi-array SXR and mode identifiability

### ISSUE THREAD

**Issue by HongSik-Yun-Fusion** [VEST-Tokamak/vaft#1001]
## Background

#991 modernizes the original VEST magnetic-island rotational-tomography work as a PyTorch-based VAFT-native model using the original two SXR arrays:

~~~text
vertical   : 20 LOS
horizontal : 20 LOS
-------------------
baseline   : 40 LOS
~~~

This baseline corresponds to the original scientific problem: sparse poloidal SXR measurements are augmented by the time evolution of a rotating magnetic island to form a rotational sinogram for 2-D reconstruction.

Current VAFT soft_x_rays mapping also contains additional arrays:

~~~text
bottom   : 16 physical LOS
lowermid : 16 physical LOS
~~~

These additional arrays carry paired Be/Al two-filter measurements. They therefore provide potentially useful information beyond simply increasing the anonymous channel count:

~~~text
additional viewing geometry
        +
filter / spectral information
~~~

This issue should investigate how these additional arrays can extend #991 into **multi-array / multi-view rotational tomography**, and whether they improve magnetic-island identifiability.

The goal is not to replace the original 40-LOS vaft_v0, but to establish the scientific basis for later vaft_v1+ revisions.

---

### Spatial identifiability

Do the additional LOS improve reconstruction of:

- poloidal mode number m;
- rational-surface localization;
- island radial scale / width;
- gross MI morphology?

### Toroidal identifiability

The original study identified an important degeneracy in a single poloidal SXR view:

~~~text
toroidal mode number n
        ↔
temporal repetition / rotation count
~~~

and therefore fixed n.

If the additional arrays are confirmed to observe the plasma from a distinct toroidal location, relative phase between views may provide independent information about n.

The central question is:

> Can the additional SXR viewing geometry reduce the original n/rotation degeneracy and make m/n structure more identifiable?

This must be demonstrated rather than assumed.

---

## 2. #746 geometry validation is prerequisite for multi-view claims

The toroidal placement of the bottom and lowermid arrays is currently provisional under #746.

Before interpreting these arrays as an independent toroidal view, confirm:

- physical port;
- IMAS phi;
- array orientation;
- LOS direction;
- relative toroidal separation.

Until then, treat them only as additional measured SXR arrays.

Do not assume:

~~~text
additional array == independent toroidal view
~~~

and do not enable n reconstruction before the geometry is validated.

---

## 3. Do not append additional channels anonymously

Avoid defining the extended input as simply:

~~~text
channel 0
channel 1
...
channel 71
~~~

Each measurement should preserve physical identity.

At minimum retain:

~~~text
array identity
physical LOS identity
R/Z geometry
toroidal angle
filter material
filter thickness
calibration
DAQ provenance
channel validity
~~~

Conceptually:

~~~text
measurement
    ├── LOS geometry
    ├── time signal
    ├── array identity
    └── spectral/filter identity
~~~

---

## 4. Be/Al pairs are not independent spatial LOS

The bottom and lowermid measurements are paired Be/Al filtered observations on the same physical LOS.

Therefore Be and Al must not be counted as two independent spatial sightlines.

Prefer a representation such as:

~~~text
physical LOS
    ↓
[time, filter/spectral feature]
~~~

or:

~~~text
[LOS, time, filter]
~~~

For example:

~~~python
SXRMeasurementSpec(
    array="bottom",
    los=5,
    filters=("Be", "Al"),
)
~~~

The contract should remain extensible to future filter / energy-response information.

---

## 5. Keep #991 vaft_v0 as the stable baseline

#991 should remain:

~~~text
vaft_v0

vertical 20
+
horizontal 20
        ↓
40-LOS rotational tomography
~~~

The first multi-array extension should be a separate model revision.

Possible lineage:

~~~text
vaft_v0
    original 40-LOS baseline

vaft_v1
    multi-array / multi-view rotational tomography

vaft_v2
    multi-view + spectral/two-filter conditioning
~~~

Do not silently redefine vaft_v0 when additional arrays are introduced.

---

## 6. Add information in controlled stages

Evaluate at least the following configurations separately.

### A. Baseline

~~~text
vertical + horizontal
~~~

### B. Spatial multi-array

~~~text
vertical
horizontal
bottom
lowermid

using additional spatial information only
~~~

If necessary, reduce each Be/Al pair to one calibrated spatial observable for this ablation.

### C. Full multi-spectral input

~~~text
vertical
horizontal
bottom(Be, Al)
lowermid(Be, Al)
~~~

This should separate performance gains caused by:

~~~text
more LOS geometry
~~~

from gains caused by:

~~~text
spectral/filter information
~~~

---

## 7. Extend the #886 synthetic forward model

Use #886 as the source of synthetic physics.

Generate all relevant SXR-array responses from the same equilibrium / MI state:

~~~text
equilibrium
    ↓
MI topology
    ↓
emissivity model
    ↓
array-specific SXR projection
~~~

For broadband/proxy channels the basic response is:

~~~math
S_i = \int_{LOS_i} \epsilon\,dl.
~~~

For two-filter measurements, a more complete future response may require:

~~~math
S_{i,f}
=
\int_{LOS_i} dl
\int dE\,
R_f(E)\,
\epsilon(E,R,Z),
~~~

where R_f(E) is the filter / detector response.

A complete spectral-radiation model is not required for the first geometry-ablation study.

---

## 8. Initial use of two-filter information

Do not immediately turn the extended model into a joint emissivity / temperature inversion.

First compare:

### Option A — auxiliary spatial measurement

Reduce or combine the Be/Al pair into a calibrated spatial observation or auxiliary feature.

### Option B — explicit filter dimension

Keep:

~~~text
LOS
 ├── Be
 └── Al
~~~

as distinct filter measurements attached to one physical LOS.

Evaluate whether preserving the spectral pair improves reconstruction beyond additional geometry alone.

---

## 9. Reuse existing two-filter Te processing

VAFT already contains two-filter SXR ratio / electron-temperature processing.

A later extension should compare:

~~~text
raw Be/Al conditioning
~~~

against:

~~~text
physics-derived two-filter Te proxy conditioning
~~~

This keeps learned representation and explicit diagnostic physics distinguishable.

---

## 10. Prepare the architecture for measurement identity

Retain the #991 Transformer architecture family but avoid over-coupling it to one fixed anonymous channel count.

Long-term, each measurement/token may incorporate:

~~~text
signal embedding
+
time embedding
+
LOS geometry embedding
+
array embedding
+
filter embedding
~~~

Conceptually:

~~~text
token
=
measurement feature
+
geometry metadata
+
diagnostic metadata
~~~

The first implementation may still use a fixed tensor layout, but public contracts should not prevent migration toward geometry-aware measurement tokens.

---

## 11. Multi-view phase information

If the additional arrays are confirmed to occupy distinct toroidal locations, explicitly investigate relative phase.

For a magnetic island:

~~~math
\xi = m\theta^* - n\phi - \alpha(t).
~~~

For two views at phi_1 and phi_2:

~~~math
\Delta \xi = -n(\phi_2-\phi_1).
~~~

This may provide information about n that is unavailable from one poloidal view alone.

Treat this as an identifiability problem, not merely another neural-network benchmark.

---

## 12. Do not add n as a target automatically

Only enable experimental n inference after all of the following are satisfied:

- [ ] #746 confirms toroidal geometry.
- [ ] The synthetic forward model uses the actual toroidal placement.
- [ ] Different n values produce distinguishable synthetic observations.
- [ ] Distinction remains under realistic measurement noise/error.
- [ ] Independent experimental MHD diagnostics can validate the result.

Until then:

~~~text
n = prescribed / conditioning synthetic parameter
~~~

rather than a reconstructed quantity.

---

## 13. Cross-diagnostic Mirnov validation

If multi-view SXR eventually supports n inference, compare it independently against Mirnov phase-array results.

~~~text
SXR-inferred n
        ↕
Mirnov-derived n
~~~

Agreement and disagreement are both scientifically useful.

Do not use Mirnov-derived n automatically as supervised SXR training truth.

---

## 14. Evaluation matrix

At minimum compare:

| Input | Purpose |
|---|---|
| vertical + horizontal | #991 40-LOS baseline |
| baseline + additional spatial LOS | geometry-information gain |
| baseline + explicit Be/Al measurements | spectral-information gain |
| confirmed multi-view geometry | toroidal-information gain |

Primary metrics:

~~~text
image MSE / MAE
SSIM
rational-surface localization error
island-width error
m identification
LOS reprojection residual
~~~

If geometry supports it, additionally evaluate:

~~~text
n identification
m/n classification
relative inter-view phase reconstruction
~~~

---

## 15. Missing-array robustness

Real shots will not always contain every SXR array or every usable channel.

The model / preprocessing contract should support:

~~~text
original 40 only
40 + one additional array
all arrays
individual bad channels
missing filter member
~~~

Use explicit measurement / array / channel-validity masks or an equivalent contract.

Do not interpret missing measurements as ordinary zero-valued channels.

---

## 16. Geometry revision is part of the ModelArtifact

Changing SXR geometry changes the inverse problem.

Record at minimum:

~~~text
array set
channel mapping
LOS geometry revision
toroidal geometry revision
filter metadata revision
~~~

A model trained before and after the #746 geometry resolution may require a new model revision even if the neural architecture is unchanged.

---

## 17. Relationship to #991

#991 — Implement a PyTorch VAFT-native rotational SXR tomography model

Direct upstream.

~~~text
#991
original vertical + horizontal 40 LOS
        ↓
PyTorch rotational tomography
        ↓
vaft_v0
~~~

This issue:

~~~text
additional SXR arrays
        ↓
multi-array / multi-view input
        ↓
information-gain / identifiability study
        ↓
future vaft_v1+
~~~

Keep v0 as a stable baseline.

---

## 18. Relationship to #886

#886 — Add equilibrium-aware magnetic-island phantom and synthetic SXR forward model

All additional-array synthetic measurements should use the same forward-model infrastructure.

Do not introduce a second LOS projector or independent MI phantom inside the ML code.

Any required spectral-response extension should be coordinated with #886 or a dedicated forward-model follow-up.

---

## 19. Relationship to #267

#267 — Add SXR tomography inversion with flux-aligned and iso-flux reconstruction

Share expanded LOS geometry / forward operators where practical.

For the same multi-array geometry, compare:

~~~text
classical tomography
vs
learned rotational tomography
~~~

when scientifically meaningful.

---

## 20. Relationship to #677

#677 — Add common measurement-error models for synthetic diagnostics

Reuse its generic infrastructure for gain uncertainty, offset, noise, and channel variation.

---

## 21. Relationship to #746

#746 — Confirm the provisional toroidal placements written for SXR, CES and the camera

This is the direct prerequisite for any claim that bottom/lowermid provide an independent toroidal view.

Until #746 is resolved, do not claim multi-toroidal n identifiability.

---

### Phase 1 — geometry and measurement audit

- [ ] Confirm physical LOS identities for vertical/horizontal/bottom/lowermid.
- [ ] Confirm Be/Al pairing semantics.
- [ ] Audit array-specific calibration / validity.
- [ ] Incorporate the final #746 geometry.
- [ ] Define multi-array input metadata.

### Phase 2 — expanded synthetic diagnostic

- [ ] Extend #886 projection to the additional arrays.
- [ ] Generate all array signals from the same synthetic MI case.
- [ ] Define the Be/Al response representation.
- [ ] Preserve array/filter provenance.
- [ ] Produce expanded SyntheticDatasetArtifacts.

### Phase 3 — spatial-information ablation

- [ ] Train/evaluate the 40-LOS baseline.
- [ ] Add additional spatial LOS.
- [ ] Compare rational-surface localization.
- [ ] Compare island scale / morphology.
- [ ] Compare forward residuals.

### Phase 4 — spectral-information ablation

- [ ] Represent Be/Al as an explicit filter dimension.
- [ ] Compare raw pair vs reduced spatial representation.
- [ ] Compare two-filter Te proxy conditioning.
- [ ] Separate geometry gain from spectral gain.

### Phase 5 — multi-view identifiability

- [ ] Use confirmed toroidal geometry.
- [ ] Perform synthetic n sensitivity scans.
- [ ] Revisit the rotation-frequency / n degeneracy.
- [ ] Evaluate identifiability with realistic noise.
- [ ] Attempt n reconstruction only if distinguishability is demonstrated.

### Phase 6 — model revision

- [ ] Finalize multi-array model input.
- [ ] Support measurement/array masks.
- [ ] Define missing-array behavior.
- [ ] Introduce vaft_v1 or the appropriate model revision.
- [ ] Record geometry/filter revisions in ModelArtifact.

### Phase 7 — experimental validation

- [ ] Select representative multi-array VEST shots.
- [ ] Compare reconstruction quality against #991 baseline.
- [ ] Compare measured vs reprojected SXR.
- [ ] Compare SXR mode evidence independently with Mirnov.
- [ ] Evaluate reconstruction stability under array subsets.

---

## Acceptance criteria

- [ ] Additional SXR arrays are not treated as anonymous extra channels.
- [ ] Physical LOS and Be/Al filter identity are preserved explicitly.
- [ ] #991 40-LOS model remains the baseline.
- [ ] Geometry and spectral information gains are evaluated separately.
- [ ] All arrays for one synthetic case derive from the same physical MI state.
- [ ] Geometry / forward operators are shared with #886/#267 where practical.
- [ ] Missing arrays/channels remain distinct from zero-valued measurements.
- [ ] Geometry and filter revisions are preserved in dataset/model provenance.
- [ ] Additional arrays are not claimed as independent toroidal views before #746 is resolved.
- [ ] n reconstruction is attempted only after synthetic identifiability is demonstrated.
- [ ] Improvements over the 40-LOS baseline are quantified.
- [ ] Experimental validation includes SXR reprojection and independent MHD evidence.

---

## Non-goals

The initial work does not require:

- assuming the additional arrays automatically determine n;
- treating Be and Al as separate spatial sightlines;
- full atomic spectral inversion;
- simultaneous full Te(R,Z) + emissivity tomography;
- 3-D emissivity tomography;
- nonlinear MHD reconstruction;
- replacing #991 vaft_v0;
- using Mirnov-derived mode numbers as supervised SXR ground truth;
- making multi-toroidal-view claims before geometry is confirmed.

---

## Long-term direction

~~~text
vaft_v0
original 40-LOS rotational tomography
        ↓
additional SXR geometry validation
        ↓
multi-array spatial tomography
        ↓
multi-view identifiability study
        ↓
spectral/two-filter conditioning
        ↓
vaft_v1 / vaft_v2
~~~

The long-term objective is:

~~~text
multiple spatial views
        +
rotational temporal information
        +
filter / spectral information
        ↓
better-constrained magnetic-island inverse problem
~~~

In particular, if the additional arrays are confirmed to provide sufficient toroidal separation, a central goal is to test whether they can reduce the n / rotation-repetition degeneracy that is structurally unresolved by the original single-poloidal-view rotational-tomography formulation.
